src/main/java/core/activities/SupplyActivitiesHandler.java

```diff
@@ -0,0 +1,11 @@
+package core.activities;
+
+public class SupplyActivitiesHandler implements ActivitiesHandler {
+    @Override
+    public int getAmountOfFruits(int fruitsInShop, int fruitsInCirculation) {
+        if (fruitsInCirculation < 0) {
+            throw new RuntimeException("Suppliers can't take negative value of fruits!");
+        }
```

Comment:
this check is the same for all 4 operations - maybe we should check for negative operand while parsing the line?